services/vehicle/src/test/java/com/wedriveu/services/vehicle/entity/VehicleStoreImplTest.java

```diff
@@ -0,0 +1,265 @@
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private boolean thereAreNotDuplicates(List<Vehicle> vehicles) {
+        for (int j = 0; j < vehicles.size(); j++) {
+            String currentCarLicencePlate = vehicles.get(j).getCarLicencePlate();
+            for (int i = 0; i < vehicles.size(); i++) {
+                if (j != i
+                        && i < vehicles.size() - 1
+                        && vehicles.get(i).getCarLicencePlate().equals(currentCarLicencePlate)) {
+                    Log.log("There are two duplicates car licence plate: "
+                            + vehicles.get(i).getCarLicencePlate()
+                            + " "
+                            + currentCarLicencePlate);
+                    return false;
+                }
+            }
+        }
+        Log.log("NO DUPLICATES!");
+        return true;
+    }
+
+    private Vehicle createVehicle(String carLicencePlate,
+                                  String state,
+                                  Position position,
+                                  Date lastUpdate) {
+        return new Vehicle(carLicencePlate, state, position, lastUpdate);
+    }
+
+    private void writeJsonVehicleFile(ArrayList<Vehicle> vehicleListToJSon) {
+        ObjectMapper mapper = new ObjectMapper();
+        checkDuplicatesAndWriteOnVehiclesDb(vehicleListToJSon, mapper);
+    }
+
+    private Vehicle getVehicleRequestedCheckingVehiclesList(List<Vehicle> vehicles, String carLicencePlate) {
+        for (Vehicle vehicle : vehicles) {
+            if(vehicle.getCarLicencePlate().equals(carLicencePlate)){
+                Log.log("com.wedriveu.services.vehicle.entity.Vehicle found! -> " +
+                        vehicle.getCarLicencePlate() +
+                        " " +
+                        vehicle.getState());
+                return vehicle;
+            }
+        }
+        Log.log("com.wedriveu.services.vehicle.entity.Vehicle not found, retry!");
+        return null;
+    }
+
+    private List<Vehicle> readFromVehiclesDb(ObjectMapper mapper){
+        try {
+            ClassLoader classLoader = getClass().getClassLoader();
+            File file = new File(classLoader.getResource("vehicles.json").getFile());
+            List<Vehicle> vehicles =
+                    mapper.readValue(file, new TypeReference<List<Vehicle>>(){});
+            return vehicles;
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    private void checkDuplicatesAndWriteOnVehiclesDb(List<Vehicle> vehicles, ObjectMapper mapper) {
+        if(thereAreNotDuplicates(vehicles)) {
+            try {
+                ClassLoader classLoader = getClass().getClassLoader();
+                File file = new File(classLoader.getResource("vehicles.json").getFile());
+                mapper.writeValue(file, vehicles);
+                String jsonInString = mapper.writeValueAsString(vehicles);
```

Comment:
The writeValue method write the vehicles list in the file.json. The writeValueAsString method gives the content of the list in string so we can log it and check what the file contains.